src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -218,36 +365,52 @@ private boolean isCompleteInt(Stack<Character> spellingStack) {
      * checks if string contains valid characters
      * <p>
      * String constants start and end with double quotes.
-     * They may contain the following special symbols:
-     * \n (newline),
-     * \t (tab),
-     * \" (double quote),
-     * \\ (backslash), and
-     * \f (form feed).
-     * A string constant cannot exceed 5000 characters and cannot span multiple lines.
      *
      * @param spellingStack the stack containing the characters
      * @return returns true if the stack contains a complete string
      */
     private boolean isCompleteString(Stack<Character> spellingStack) {
-        // todo: if EOF is the last char then this is an terminated string
-        // todo: check for illegal characters
         int stackSize = spellingStack.size();
-        if (stackSize > 1 && stackSize <= 5000) {
-            if (spellingStack.peek() == '"' && spellingStack.get(stackSize - 2) != '\\') {
+        if (stackSize > 1) {
+            // check if string is closed
+            if (spellingStack.peek() == '"' && !charIsEscaped) {
```

Comment:
@KitchenTable99  !charIsEscaped may actually be better because it avoids the look up for the second to top element?

Reply:
Yea that makes sense